### PR TITLE
Feat(e2e): Components tests

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.azure.digitaltwins.core.TestHelper.DISPLAY_NAME_WITH_ARGUMENTS;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ComponentsTests extends ComponentsTestBase {
+
+    private final ClientLogger logger = new ClientLogger(ModelsTestBase.class);
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java:24]: new ClientLogger(**ComponentsTests**.class)

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+            logger.info("Created {} twin successfully", createdTwin.getId());
+            assertEquals(createdTwin.getId(), roomWithWifiTwinId);
+
+            // Get the component
+            Response<String> getComponentResponse = client.getComponentWithResponse(roomWithWifiTwinId, wifiComponentName, Context.NONE);
+            assertEquals(getComponentResponse.getStatusCode(), HttpURLConnection.HTTP_OK);
+
+            // Update component
+            DigitalTwinsResponse<Void> upDateComponentResponse = client.updateComponentWithResponse(
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java:59]: nit: "update" not "upDate"

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    client.deleteModel(roomWithWifiModelId);
+                }
+                if (wifiModelId != null)
+                {
+                    client.deleteModel(wifiModelId);
+                }
+            }
+            catch (Exception ex)
+            {
+                fail("Test clean up failed: " + ex.getMessage());
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java:87]: To preserve the full stacktrace, how about:

throw new AssertionFailedError("Test clean up failed", ex)

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java [hunk 4]
```diff
@@ -72,6 +72,16 @@ protected DigitalTwinsClientBuilder getDigitalTwinsClientBuilder(HttpClient http
         return builder;
     }
 
+    protected DigitalTwinsClient getClient(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) {
+        return getDigitalTwinsClientBuilder(httpClient, serviceVersion)
+            .buildClient();
+    }
+
+    protected DigitalTwinsAsyncClient getAsyncClient(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) {
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java:80]: good idea moving this up to a higher level

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsAsyncTests.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                    asyncClient.deleteModel(roomWithWifiModelId).block();
+                }
+                if (wifiModelId != null)
+                {
+                    asyncClient.deleteModel(wifiModelId).block();
+                }
+            }
+            catch (Exception ex)
+            {
+                fail("Test clean up failed: " + ex.getMessage());
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsAsyncTests.java:88]: Try to preserve the stacktrace here, too

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java (hunk 2 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java:59]: :)) good catch !

**Review by timtay-microsoft — APPROVED** [review]
Just a few small comments, but otherwise looks good

**Pull request merged**